tests/test_code_analyzer.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+import threading
+
+from bionic.code_analyzer import CodeHasher, CodeHashPrefix
+
+
+def test_code_hasher():
+    def barray(value):
+        return bytearray(value, "utf8")
+
+    circular_dict = {"k1": "v1"}
+    circular_dict["k2"] = circular_dict
+
+    def f1():
+        v = 10
+        return v
+
+    def f2():
+        v = 20
+        return v
+
+    def inc(x):
+        return x + 1
+
+    def dec(x):
+        return x - 1
+
+    def r1():
```

Comment:
Isn't this the same as `f1`, but with a different name?

Reply:
I realized that after adding to the `idx_by_hash_value` dict.